modules/index-service/src/main/java/org/opencastproject/index/service/impl/index/event/Event.java

```diff
@@ -1150,52 +1044,29 @@ private void updateEventStatus() {
 
     if (getWorkflowId() != null && StringUtils.isNotBlank(getWorkflowState())) {
       eventStatus = workflowStatusMapping.get(getWorkflowState());
-      return;
-    }
-
-    if (StringUtils.isNotBlank(getRecordingStatus())) {
+    } else if (StringUtils.isNotBlank(getRecordingStatus())) {
       eventStatus = recordingStatusMapping.get(getRecordingStatus());
-      return;
-    }
-
-    if (BooleanUtils.isTrue(getOptedOut())) {
-      eventStatus = "EVENTS.EVENTS.STATUS.OPTEDOUT";
-      return;
-    }
-
-    if (StringUtils.isNotBlank(getSchedulingStatus())) {
+    } else if (isScheduledEvent()) {
       eventStatus = "EVENTS.EVENTS.STATUS.SCHEDULED";
-      return;
+    } else {
+      /* This can be the case if all workflows of an event have been deleted */
+      eventStatus = "EVENTS.EVENTS.STATUS.PROCESSED";
     }
-
-    eventStatus = "EVENTS.EVENTS.STATUS.PROCESSED";
   }
 
-  /**
-   * Check whether or not the start time of the recording of this event has passed, yet.
-   *
-   * This method treats non-scheduled (i.e.) uploaded events as always having
-   * a recording start time in the pase, because for the video to be able
-   * to be uploaded, it obviously must have been recorded, already.
-   *
-   * @return <code>true</code> if the start ime of the recording of this event
-   *         is in the past, and <code>false</code> otherwise
-   */
-  public boolean hasRecordingStarted() {
-    return
-            // handle uploaded events
-            getSchedulingStatus() == null
-            // handle scheduled events
-            || getRecordingStatus() != null;
+  public boolean isScheduledEvent() {
+    /* Only scheduled events have a capture ID assigned */
+    return StringUtils.isNotBlank(getAgentId());
```

Comment:
@lkiesow Instead of using optOut to determine whether an event has scheduling information, we now just use the capture agent ID (this is not the same field as "location"/"room")...